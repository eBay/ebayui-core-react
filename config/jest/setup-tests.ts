import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
