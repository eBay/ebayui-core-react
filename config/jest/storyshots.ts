import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
