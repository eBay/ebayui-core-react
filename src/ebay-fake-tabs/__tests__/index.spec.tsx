import React from 'react'
import { initStoryshots } from '../../../config/jest/storyshots'

initStoryshots({
    config: ({ configure }) =>
        configure(() => {
            require('./index.stories')
        }, module)
})
