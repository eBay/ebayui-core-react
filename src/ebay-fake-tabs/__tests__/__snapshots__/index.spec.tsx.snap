// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-fake-tabs Default Tabs 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a
          aria-current="page"
          href="http://ebay.com"
        >
          eBay.com
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          href="http://ebay.de"
        >
          eBay.de
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          href="http://ebay.co.uk"
        >
          eBay.co.uk
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__cell"
      >
        <h3>
          eBay.com Content
        </h3>
        <p>
          Some US content...
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-fake-tabs Pre-selected Tab 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a
          href="http://ebay.com"
        >
          eBay.com
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          aria-current="page"
          href="http://ebay.de"
        >
          eBay.de
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          href="http://ebay.co.uk"
        >
          eBay.co.uk
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__cell"
      >
        <h3>
          eBay.de Content
        </h3>
        <p>
          Some German content...
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Storyshots ebay-fake-tabs tabMatchesCurrentUrl: false 1`] = `
<DocumentFragment>
  <div
    class="fake-tabs"
  >
    <ul
      class="fake-tabs__items"
    >
      <li
        class="fake-tabs__item"
      >
        <a
          href="http://ebay.com"
        >
          eBay.com
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          href="http://ebay.de"
        >
          eBay.de
        </a>
      </li>
      <li
        class="fake-tabs__item"
      >
        <a
          aria-current="true"
          href="http://ebay.co.uk"
        >
          eBay.co.uk
        </a>
      </li>
    </ul>
    <div
      class="fake-tabs__content"
    >
      <div
        class="fake-tabs__cell"
      >
        <h3>
          eBay.co.uk Content
        </h3>
        <p>
          Some British content...
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
